docs/architecture.md

```diff
@@ -0,0 +1,114 @@
+- Posts Service (`src/api/posts`) - API for accessing Post and Feed data in Redis (probably not well named at this point)
+- Search Service (`src/api/search`) - API for doing searches against Elasticsearch
+- Status Service (`src/api/status`) - API for accessing Telescope status information, as well as providing the Dashboards
+- Users Service (`src/api/users`) - API for interacting with the Users data (currently in Firebase)
+- Parser Service (`src/api/parser) - feed and post parsing. Currently disabled, see <https://github.com/Seneca-CDOT/telescope/issues/2111>
+
+All microservices are built on a common foundation, the [Satellite module](https://github.com/Seneca-CDOT/satellite). Satellite provides a common set of features for building Express-based microservices, with proper logging, health checks, headers, authorization middleware, as well as connections to Redis and Elasticsearch. It saves us having to manage the same set of dependencies a dozen times, and repeat the same boilerplate code.
+
+The microservices are deployed as Docker containers. Each defines its own `Dockerfile`, and the entire system is managed via `docker-compose`, see `src/docker`, where we have a base service definition in `src/docker/docker-compose.yml` as well as overrides for various environments: `development.yml` (local dev), `production.yml` (staging and production), and `gitpod.yml` (running Telescope in GitPod). Each of these service definitions uses an associated environment file, which are defined in `src/config`. For example, Telescope's production server uses the `src/docker/production.yml` file and `src/config/env.production` environment.
+
+When deployed, the services are not directly accessible via the network (i.e. you can't go to localhost:5555 to access the server). Instead, you have to access them via an API gateway called Traefik, which also runs in its own container. Traefik lives at api.telescope.cdot.systems (production), dev.api.telescope.cdot.systems (staging), and localhost:80 (development). It provides routing to each of the service containers via labels defined in the `src/docker/docker-compose.yml` file. For example, the Image Service is available at `/v1/image`. In production you access it at <https://api.telescope.cdot.systems/v1/image>, in staging at <https://dev.api.telescope.cdot.systems/v1/image>, and locally at <http://localhost/v1/image>. The other services are similar. Traefik makes the different services look like one thing, but they are actually deployed and running separately.
+
+The microservices can be controlled either using the `docker-compose` command directly, or with the following `npm` scripts:
+
+- `npm run services:start` starts all services. You can also list one or more services to start a subset, `npm run services:start image planet`
+- `npm run services:stop` stops all services, or those specified
+- `npm run logs image` shows logs for the image service (specify whichever service you want to view)
```

Comment:
`pnpm` instead of `npm`?

Reply:
Yup, fixed all occurrences after you spotted the first.  New commits have updates.